src/System.Security.Cryptography.Algorithms/tests/CryptoConfigTests.cs

```diff
@@ -234,6 +236,57 @@ public static void CreateFromName_CtorArguments()
             Assert.Equal("Hello", ctorObj.MyString);
         }
 
+        [Fact]
+        public static void EncodeOID_Validation()
+        {
+            Assert.Throws<ArgumentNullException>(() => CryptoConfig.EncodeOID(null));
+            Assert.Throws<FormatException>(() => CryptoConfig.EncodeOID(string.Empty));
+            Assert.Throws<FormatException>(() => CryptoConfig.EncodeOID("BAD.OID"));
+            Assert.Throws<FormatException>(() => CryptoConfig.EncodeOID("1.2.BAD.OID"));
+            Assert.Throws<OverflowException>(() => CryptoConfig.EncodeOID("1." + uint.MaxValue));
+            CryptoConfig.EncodeOID("-1.2.-3"); // Negative values not checked
+        }
+
+        [Fact]
+        public static void EncodeOID_Length_Boundary()
+        {
+            string valueToRepeat = "1.1";
+
+            // Max size is 0x7f
+            string s = new StringBuilder(valueToRepeat.Length * 0x80).Insert(0, valueToRepeat, 0x80).ToString();
+            Assert.Throws<CryptographicUnexpectedOperationException>(() => CryptoConfig.EncodeOID(s));
+
+            s = new StringBuilder(valueToRepeat.Length * 0x7f).Insert(0, valueToRepeat, 0x7f).ToString();
+            CryptoConfig.EncodeOID(s);
+        }
+
+        [Theory]
+        [InlineData(0x4000, "0603818028")]
+        [InlineData(0x200000, "060481808028")]
+        [InlineData(0x10000000, "06058180808028")]
+        [InlineData(0x10000001, "06058180808029")]
+        [InlineData(int.MaxValue, "060127")]
```

Comment:
All of these produce wrong answers, in that `0.` and `1.` are restricted to the second value being in the 0..39 range.

int.MaxValue produces a REALLY wrong answer, though, and definitely needs to be in a "compat" test.

If it was `2.2147483647` (instead of 1.2147483647) the correct answer is `06 05 88 80 80 80 4f`.  So, for an invalidly encoded 1. entry it should be coming out as `06 05 88 80 80 80 27` (if my math is right).

The compat tests serve a good purpose: They tell me (or anyone else, I suppose) that we probably can't fix this method to properly encode OIDs, because not only does it throw on legal ones, it returns wrong answers sometimes (mathematically, not just "by policy").